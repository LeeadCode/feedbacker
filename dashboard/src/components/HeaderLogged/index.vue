<template>
  <div class="flex items-center justify-between w-full md:w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36">
      <img class="w-full" src="../../assets/images/logo_white.png" alt="Logo Feedbacker">
    </div>

    <div class="flex">
      <ul class="flex list-none">
        <li
          @click="() => router.push({ name: 'Credentials' })"
          class="links text-white"
        >
          Credentials
        </li>

        <li
          @click="() => router.push({ name: 'Feedbacks' })"
          class="links text-white"
        >
          Feedbacks
        </li>

        <li
          @click="handleLogout"
          class="links bg-white text-brand-main"
          id="logout-button"
        >
          {{ logoutLabel }}
        </li>
      </ul>
    </div>
  </div>
</template>

<script setup>
import useStore from '@/hooks/useStore';
import { clearCurrentUser } from '@/Store/user';
import { computed } from 'vue';
import { useRouter } from 'vue-router';

const router = useRouter()
const store = useStore('User')

const logoutLabel = computed(() => {
  if(!store.currentUser.name){
    return '...'
  }

  return `${store.currentUser.name} (sair)`
})

function handleLogout () {
  clearCurrentUser()
  router.push({name: 'Home'})
}

</script>

<style lang="pcss" scoped>
.links{
    @apply px-6 py-2 font-bold focus:outline-none rounded-full cursor-pointer;
}
</style>