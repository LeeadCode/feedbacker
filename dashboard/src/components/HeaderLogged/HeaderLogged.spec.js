import { shallowMount } from '@vue/test-utils'
import HeaderLogged from '.'
import { routes } from '../../router'

import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

const mockStore = { currentUser: {} }
jest.mock('../../hooks/useStore', () => {
  return () => mockStore
})

async function customBeforeEach(fn) {
  router.push('/')
  await router.isReady()

  if(fn) fn()

  return(shallowMount(HeaderLogged, {
    global: {
      plugins: [router]
    }
  }))
}

describe('<HeaderLogged />', () => {
  it('should render header logged correctly', async () => {
    const wrapper = await customBeforeEach()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 3 dots when theres not user logged', async () => {
    const wrapper = await customBeforeEach()
    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('...')
  })

  it('should render a user name when user logged', async () => {
    const wrapper = await customBeforeEach(() => mockStore.currentUser.name = 'Igor')
    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('Igor (sair)')
  })
})