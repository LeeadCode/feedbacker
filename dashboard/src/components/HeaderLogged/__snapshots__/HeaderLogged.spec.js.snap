// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render header logged correctly 1`] = `
<div class="flex items-center justify-between w-full md:w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full" src="" alt="Logo Feedbacker"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li class="links text-white"> Credentials </li>
      <li class="links text-white"> Feedbacks </li>
      <li class="links bg-white text-brand-main" id="logout-button">...</li>
    </ul>
  </div>
</div>
`;
